src/com/untamedears/realisticbiomes/listener/PlayerListener.java

```diff
@@ -86,102 +91,138 @@ public PlayerListener(RealisticBiomes plugin, Map<Object, GrowthConfig> growthCo
+				int data = event.getItem().getData().getData();
+				if (saplingIndexMap.containsKey(data)) {
+					material = saplingIndexMap.get(data);
 				}
-				
-				
-				// don't do anything if the material is a dye, but not cocoa
-				if (event.getMaterial() == Material.INK_SACK && event.getItem().getData().getData() != 3/*cocoa*/)
+			}
+			
+//			RealisticBiomes.doLog(Level.FINER, "CLICKY: " + event.getItem().getData());
+			
+			// don't do anything if the material is a dye, but not cocoa brown
+			if (event.getMaterial() == Material.INK_SACK) {
+				MaterialData data = event.getItem().getData();
+				if ((data instanceof Dye) && ((Dye)data).getColor() != DyeColor.BROWN) {
 					return;
-				
-				block = block.getRelative(0,1,0);
+				}
 			}
-			else if (event.getAction() == Action.RIGHT_CLICK_BLOCK && (material == Material.STICK || material == Material.BONE)) {
-				// right click on a growing crop with a stick: get information about that crop
-				material = event.getClickedBlock().getType();
+			
+		}
+		else if (event.getAction() == Action.RIGHT_CLICK_BLOCK && (material == Material.STICK || material == Material.BONE)) {
+			// right click on a growing crop with a stick: get information about that crop
+			material = event.getClickedBlock().getType();
+			
+			// handle saplings as their tree types
+			int index = block.getData();
+			if (material == Material.SAPLING && saplingIndexMap.containsKey(index)) {
 				
-				// handle saplings as their tree types
-				int index = block.getData();
-				if (material == Material.SAPLING && saplingIndexMap.containsKey(index)) {
-					
-					material = saplingIndexMap.get(index);
-				}
+				material = saplingIndexMap.get(index);
+			}
+			
+			GrowthConfig growthConfig = growthConfigs.get(material);
+			if (plugin.persistConfig.enabled && growthConfig != null && growthConfig.isPersistent()) {
 				
-				GrowthConfig growthConfig = growthConfigs.get(material);
-				if (plugin.persistConfig.enabled && growthConfig != null && growthConfig.isPersistent()) {
-					
-					plantGrowth = plugin.growAndPersistBlock(block, false, growthConfig);
-				}
+				plant = plugin.growAndPersistBlock(block, false, growthConfig, null);
 			}
-			else {
-				// right clicked without stick, do nothing
-				return;
+		}
+		else {
+			// right clicked without stick or bone, do nothing
+			return;
+		}
+		
+		// show growth rate information if the item in the player's hand is not a bone
+		if (event.getMaterial() == Material.BONE) {
+			return;
+		}
+		
+		block = event.getClickedBlock();
+		if (event.getAction() == Action.LEFT_CLICK_BLOCK) {
+			// from hitting something with material in hand
+			if (material == Material.COCOA) {
+				block = block.getRelative(event.getBlockFace());
+			} else {
+				block = block.getRelative(0,1,0);
 			}
+		}
+		
+		GrowthConfig growthConfig = growthConfigs.get(material);
+		if (growthConfig == null)
+			return;

```

Comment:
Ahh, usually github is nice and highlights those for me. 

I won't hold it against you if you kill it with fire. I've been knocking out the ones I find in FactoryMod.